### ISSUE TITLE
Compute predictions on training data

### ISSUE THREAD

**Issue by AndreaPi** [cornellius-gp/gpytorch#1957]
First of all, compliments for this great library! Among all the Gaussian Process libraries in Python I tried, I found GPyTorch to be _by far_ the fastest and the most accurate!

I split my dataset in `train`, `test`, and I trained my 2-outputs Gaussian Process Regression model as described here:
https://docs.gpytorch.ai/en/v1.5.1/examples/03_Multitask_Exact_GPs/Multitask_GP_Regression.html

Now, I would like to report both the training and test MSE: computing the test MSE is straightforward,

```python
import numpy as np
from sklearn.metrics import mean_squared_error

model.eval()
likelihood.eval()
with torch.no_grad(), gpytorch.settings.fast_pred_var():
    observed_pred = likelihood(model(test_x))
mean = observed_pred.mean.cpu()
test_MSE = mean_squared_error(test_y.cpu(), mean.detach().numpy(), multioutput='raw_values')
```

For training RMSE, I got a warning:

```
with torch.no_grad(), gpytorch.settings.fast_pred_var():
    observed_pred = likelihood(model(train_x))
mean = observed_pred.mean.cpu()
train_MSE = mean_squared_error(train_y.cpu(), mean.detach().numpy(), multioutput='raw_values')
```
`
/opt/conda/lib/python3.8/site-packages/gpytorch/models/exact_gp.py:273: GPInputWarning: The input matches the stored training data. Did you forget to call model.train()?`

It sounds like GPyTorch is not happy when I pass it training data. However, computing the difference between train MSE and test MSE is an important step of a ML pipeline. Am I doing something wrong? Is there another way to return the model predictions on the training data?

**Comment by patel-zeel** [comment]
Hi @AndreaPi, AFAIK this warning (intentionally a warning and not an error) is given supposing that you may have missed setting the `.train()` mode while training (A common gotcha!). In your case, this is completely normal and it should work without a problem.

P.S.: We have `gpytorch.metrics` as well (not released yet but available via the repo) if you want to get rid of `.detach()` and `.numpy()` and be in `torch` end-to-end. We would love to hear your feedback on it :)

**Comment by AndreaPi** [comment]
> Hi @AndreaPi, AFAIK this warning (intentionally a warning and not an error) is given supposing that you may have missed setting the .train() mode while training (A common gotcha!). In your case, this is completely normal and it should work without a problem.

Good to know!

> P.S.: We have `gpytorch.metrics` as well (not released yet but available via the repo) if you want to get rid of `.detach()` and `.numpy()` and be in `torch` end-to-end. It also has other probabilistic metrics that make more sense with GP models. More details are https://github.com/cornellius-gp/gpytorch/issues/1857. We would love to hear your feedback on it :)

Thanks for the suggestion! However, I need metrics to compare different models (not all in GPyTorch), thus I can't use metrics such as NLPD, MSLL or CE that can only be computed for GPyTorch models. Also, I don't know enough about these metrics to interpret them correctly, so I'll stick to my trusty MSE and R2 for now...but if you added some documentation about these metrics, what they measure and how to interpret them, I may change idea in the future 😀

**Comment by AndreaPi** [comment]
@patel-zeel @wjmaddox quick question: does this project have a Twitter handle? I'd like to tweet about my experience with GPyTorch. To quote myself:

> Among all the Gaussian Process libraries in Python I tried, I found GPyTorch to be by far the fastest and the most accurate!

**Comment by wjmaddox** [comment]
afaik, jake doesn't have twitter. but geoff (gpleiss), max (balandat) and andrew (andrewgwils) do.
